server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -233,6 +257,36 @@ private RemoteRestoreResult executeRestore(
         return RemoteRestoreResult.build(restoreUUID, restoreInfo, allocationService.reroute(updatedState, "restored from remote store"));
     }
 
+    private ClusterState restoreClusterMetadata(ClusterState restoredClusterState, Metadata clusterMetadata) {
+        Metadata.Builder mdBuilder = new Metadata.Builder();
```

Comment:
could we have used Metadata.fromXContext directly? you can pass the fields you want to restore explicitly.

Reply:
restore the metadata version as well

Reply:
should increment the version?

Reply:
`Metadata.toXContent` also stores clusterUUID, clusterUUIDCommited, and coordinationMetadata. will have to ensure these 3 are not restored from metadata